### ISSUE TITLE
Build loop with circular HTML links 

### ISSUE THREAD

**Issue by philo23** [parcel-bundler/parcel#5324]
# 🐛 bug report

Two HTML pages containing links back to each other (eg, 1.html linking to 2.html and 2.html linking back to 1.html) seems to cause an infinite build loop when you run `parcel serve 1.html` and begin editing files. It will then keep looping every time you run `parcel serve` until you delete the `.parcel-cache` folder.

Note: it will serve up fine the first time you run `parcel serve` but if you make any changes while `parcel serve` is still running (or even stop and rerun it), it will then enter the infinite loop.

`parcel build 1.html` still works as expecting without any loops though.

Also as a side note I wasn't originally expecting Parcel to build the linked HTML file, I thought I'd have to manually specify it as a second entry… but was pleasantly surprised to see it automatically pick it up!

## 🎛 Configuration (.babelrc, package.json, cli command)

See code sample below for a complete example of the bug

## 🤔 Expected Behavior

Build the two inter-linked HTML entries and stop building

## 😯 Current Behavior

"Build in [x]ms" keeps flashing with different times suggesting Parcel keeps rebuilding the two linked files back and forth.

## 💁 Possible Solution

I'm guessing something keeps updating a file parcel is watching for changes and causes another rebuild, perhaps the URL being rewritten inside of the `<a href>` for each link? No cache-breaking hashes are being added to the URLs though so it must just be file modified time if anything.

## 🔦 Context

Currently using Parcel to effectively build a static HTML site.

## 💻 Code Sample

The following steps should reproduce the bug. Note that I've cut the doctype/html/head/body tags from the HTML, but this bug is present regardless.

### package.json
```json
{
  "name": "parcel-test",
  "version": "1.0.0",
  "scripts": {
    "start": "parcel serve ./src/1.html",
    "build": "parcel build ./src/1.html"
  },
  "devDependencies": {
    "parcel": "^2.0.0-nightly.440"
  }
}
```
### src/1.html
```html
<a href="2.html">Second page</a>
```
### src/2.html
```html
<a href="1.html">Back to 1.html</a>
```
```bash
npm serve src/1.html
```
Then just edit src/1.html or src/2.html in a way that would normally trigger parcel to rebuild.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.440 |
| Node             | 14.15.0 |
| npm/Yarn         | npm 6.14.8 |
| Operating System | macOS |

**Comment by philo23** [comment]
Something else I've just noticed too that might be useful to debugging this, when I run `parcel build src/1.html` I actually end up with two hashed copied of all the external assets (css/images/js/etc) referenced in the entry HTML file.

**Comment by mischnic** [comment]
Might be the same bug as https://github.com/parcel-bundler/parcel/issues/5002 or https://github.com/parcel-bundler/parcel/issues/5355

**Comment by mischnic** [comment]
Seems fixed with 2.0.0-nightly.554.
